src/corelibs/U2Designer/src/DelegateEditors.cpp

```diff
@@ -414,6 +414,7 @@ QWidget *ComboBoxWithChecksDelegate::createEditor(QWidget *parent,
                                                   const QModelIndex & /* index */) const {
     ComboBoxWithChecksWidget *editor = new ComboBoxWithChecksWidget(items, parent);
     connect(editor, SIGNAL(valueChanged(const QString &)), this, SIGNAL(si_valueChanged(const QString &)));
+    connect(editor, SIGNAL(si_update(QWidget *)), this, SIGNAL(commitData(QWidget *)));
```

Comment:
ComboBoxWithChecksWidget already has si_valueChanged signal that should be emitted every time when something changes. Why do you need a new signal here?

Reply:
> ComboBoxWithChecksWidget already has si_valueChanged

Where?

Reply:
If my changes are stupid, why does this bug exist?

Reply:
`si_valueChanged` has the signature `void si_valueChanged(const QVariant &)`
`si_update` has the signature `void si_update(QWidget *)`

I could

1. Save the `editor` as a member of the class
2. Create a new slot for the `ComboBoxWithChecksDelegate` that `emit commitData (editor)`
3. Connect this slot to `ComboBoxWithChecksWidget::si_valueChanged`

But I can't do the first one, because `ComboBoxWithChecksDelegate::createEditor` is a const method. I am open to other ideas and discussion

Reply:
AFAIU the widget is the sender?

Reply:
> AFAIU the widget is the sender?

Yes

Reply:
ComboBoxWithChecksWidget has a correct interface: it can notify clients when something is changed using si_valueChanged.

To translate this signal into another you should listen for it in the ComboBoxWithChecksDelegate and re-emit the correct one. Check the SpinBoxDelegate::createEditor

Everything else looks good. Will approve once this part is fixed.